share/wake/lib/system/job.wake

```diff
@@ -157,6 +160,20 @@ export def getPlanKeep p =
 export def getPlanShare p =
     isShare (getPlanPersistence p)
 
+# Convenience set methods
+
+# Sets the `FnOutputs` field in a `Plan` by using the given filter function to filter out unneeded files created
+# during a job. This function is recommended over directly setting the `FnOutputs` field with `setPlanFnOutputs`.
+#
+# Parameters:
+#  - `filterFn`: Determines whether to keep a file that was created or not. True means keep the file.
+export def setPlanFilterOutputs (filterFn: (file: String) => Boolean): Plan => Plan =
+    # This implementation prevents any unintentional abuse of the FnOutputs field, which can be used to
+    # actually add files. This method here only supports filtering.
+    def fnOutputs (files: List String): List String =
+        filter filterFn files
+    setPlanFnOutputs fnOutputs
```

Comment:
I know we were talking about this in terms of *set*PlanFilterOutputs during the meeting, but I wonder if it would be better to use `editPlanFnOutputs` and so allow multiple rounds of file filtering?  I can imagine cases where the job knows that it never needs some intermediate file the tool generates on the way to the final output, some job-running combinator does something to the command which has the side effect of writing out log files but which it doesn't want to expose to Wake at large, and the final Runner produces a number of artifacts as the result of sandboxing which are obviously also low-level considerations which shouldn't be exposed.

Reply:
Yeah its a bit frustrating that `editPlanFilterOutputs` can't be implemented. Maybe we could do something like `addPlanFilterOutputs` that would be just compose FnOutputs with a filter instead.